bin/cli.js

```diff
@@ -55,12 +55,18 @@ function mkConfigObj(opts) {
 function initKnex(env, opts) {
   checkLocalModule(env);
 
-  if (!env.configuration) {
-    if (opts.client) env.configuration = mkConfigObj(opts);
-    else
+  if (!env.pathToKnexFile) {
+    if (opts.client) {
+      env.configuration = mkConfigObj(opts);
+    } else {
       exit(
-        'No knexfile found in this directory. Specify a path with --knexfile'
+        'No knexfile found in this directory. Specify a path with --knexfile or pass --client and --connection params in commandline'
```

Comment:
I though this was broken recently, but looks like it was the case for quite a while. Doesn't look like we search for "knexfile in this directory" at all. We probably should.